### PR TITLE
Update local_mods in git.py to show destination of conflict

### PR DESCRIPTION
**Pull request by ealmonte32**
specify that the local modification is in the destination directory so that some of the users clearly understand this..

##### SUMMARY

some of our users (https://github.com/Screenly/screenly-ose) a lot of times run into this error because by default ansible no longer forces the destination directory files to be replaced with the source repo which is perfectly fine, but the error message sometimes goes over their head and I couldn't find very layman's terms documentation literally saying to the users something like "you have files in the destination directory that are different than those being pulled by git from the repo you are using, please either merge your local changes with your upstream repo or backup those changes somewhere else so that it does not conflict with the current pull of files"... something like that.. I mean this is just adding one or two words to the error message that's all, it will probably go a long way saving some beginner users some headache when they make one change in their local directory and forgot that they did it and try to run ansible git and all of a sudden keep getting this local modifications error...

thanks..

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

git module

### CODE REVIEW

File: lib/ansible/modules/git.py [hunk 1]
```diff
@@ -1249,11 +1249,11 @@ def main():
         if local_mods:
             # failure should happen regardless of check mode
             if not force:
-                module.fail_json(msg="Local modifications exist in repository (force=no).", **result)
+                module.fail_json(msg="Local modifications exist in the destination directory (force=no).", **result)
```
Comment (bcoca) [inline lib/ansible/modules/git.py]: instead of rephrasing, i would just add the specific directory here

also requires a changelog

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/git.py (hunk 1 above)
Comment (ealmonte32) [reply lib/ansible/modules/git.py]: Hi @bcoca 
Thanks for the suggestion.
Changes are done, tested, it now shows the destination on the msg line.
Can you explain how I would do a changelog? I've never done that for this repo before, if you could just help me do that, thanks.

Comment (acozine) [reply lib/ansible/modules/git.py]: Hi @ealmonte32 thanks for the response (and for the PR!). We have documentation on adding changelog entries here: https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs. If you have questions that the docs don't cover, join the #ansible-docs, #ansible-community, or #ansible-dev channels on freenode IRC, we're usually hanging out there.

Comment (ealmonte32) [reply lib/ansible/modules/git.py]: Hey @acozine , thanks for the info/link 😄 
Is this what I needed to do? -> https://github.com/ansible/ansible/pull/72624

Comment (ealmonte32) [reply lib/ansible/modules/git.py]: @acozine @bcoca 
Ok sorry for the delay, it was first time having to add changelog file to already committed PR.
It's done. Let me know if that is all.
Thanks.

**Comment by ealmonte32** [comment]
Pls let me know what else this PR needs, I already did the requested changes.
Thanks.

**Comment by ealmonte32** [comment]
hmm.. one year later..
anyway, these 9 failed checks don't make sense since the code changed has nothing wrong with it at all..

**Comment by samccann** [comment]
@ealmonte32 yeah we are having problems with CI. I'll see if I can get individual tests to pass.

**Review by bcoca — APPROVED** [review]

**Pull request merged**